test/e2e/tests/test_sagemaker_endpoint_autoscaling.py

```diff
@@ -0,0 +1,201 @@
+)
+
+TARGET_RESOURCE_PLURAL = "scalabletargets"
+POLICY_RESOURCE_PLURAL = "scalingpolicies"
+ENDPOINT_STATUS_INSERVICE = "InService"
+
+
+@pytest.fixture(scope="module")
+def name_suffix():
+    return random_suffix_name("sagemaker-endpoint", 32)
+
+
+@pytest.fixture(scope="module")
+def applicationautoscaling_client():
+    return boto3.client("application-autoscaling")
+
+
+@pytest.fixture(scope="module")
+def sagemaker_endpoint(name_suffix):
+    model_name = name_suffix + "-model"
+    endpoint_config_name = name_suffix + "-config"
+    endpoint_name = name_suffix
+    resource_id = f"endpoint/{endpoint_name}/variant/variant-1"
+
+    model_input, model_response = sagemaker_make_model(model_name)
+    endpoint_config_input, endpoint_config_response = sagemaker_make_endpoint_config(
+        model_name, endpoint_config_name
+    )
+    endpoint_input, endpoint_response = sagemaker_make_endpoint(
+        endpoint_name, endpoint_config_name
+    )
+
+    wait_sagemaker_endpoint_status(endpoint_name, ENDPOINT_STATUS_INSERVICE)
+    assert resource_id is not None
+
+    yield resource_id
+    sagemaker_client().delete_endpoint(EndpointName=endpoint_name)
+    sagemaker_client().delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+    sagemaker_client().delete_model(ModelName=model_name)
+
+
+@pytest.fixture(scope="module")
+def generate_sagemaker_target(sagemaker_endpoint):
+    resource_id = sagemaker_endpoint
+    target_resource_name = random_suffix_name("sagemaker-scalable-target", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["SCALABLETARGET_NAME"] = target_resource_name
+    replacements["RESOURCE_ID"] = resource_id
+
+    target_reference, _, target_resource = create_applicationautoscaling_resource(
+        resource_plural=TARGET_RESOURCE_PLURAL,
+        resource_name=target_resource_name,
+        spec_file="sagemaker_endpoint_autoscaling_target",
+        replacements=replacements,
+    )
+
+    assert target_resource is not None
+
+    yield (resource_id, target_reference)
+
+    if k8s.get_resource_exists(target_reference):
+        _, deleted = k8s.delete_custom_resource(target_reference)
+        assert deleted
+
+
+@pytest.fixture(scope="module")
+def generate_sagemaker_policy(generate_sagemaker_target):
+    resource_id, target_reference = generate_sagemaker_target
+    policy_resource_name = random_suffix_name("sagemaker-scaling-policy", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["SCALINGPOLICY_NAME"] = policy_resource_name
+    replacements["RESOURCE_ID"] = resource_id
+
+    policy_reference, _, policy_resource = create_applicationautoscaling_resource(
+        resource_plural=POLICY_RESOURCE_PLURAL,
+        resource_name=policy_resource_name,
+        spec_file="sagemaker_endpoint_autoscaling_policy",
+        replacements=replacements,
+    )
+
+    assert policy_resource is not None
+    assert k8s.get_resource_arn(policy_resource) is not None
```

Comment:
this should be part of the test, not the fixture because this is the resource under test, it means you are already asserting create succeeded

Reply:
removed, it was duplicated.